txindex/filter.go

```diff
@@ -13,38 +13,45 @@ const (
 	TxFilterTransferred int32 = 3
 	TxFilterStaking     int32 = 4
 	TxFilterCoinBase    int32 = 5
+	TxFilterRegular     int32 = 6
 )
 
-func DetermineTxFilter(txType string, txDirection int32) int32 {
-	if txType == txhelper.TxTypeCoinBase {
-		return TxFilterCoinBase
-	}
-	if txType != txhelper.TxTypeRegular {
-		return TxFilterStaking
+func CompareTxFilter(txFilter int32, txType string, txDirection int32) bool {
```

Comment:
What is an example use case for this function? I sort of prefer the previous `DetermineTxFilter` function.

Reply:
Used to check if an incoming transaction matches currently selected filter.